fbw-common/src/wasm/systems/systems/src/structural_flex/wing_flex.rs

```diff
@@ -0,0 +1,2089 @@
+            wing_dev_damping_4_id: context.get_identifier("WING_FLEX_DEV_DAMPING_4".to_owned()),
+            neg_flex_coeff_id: context.get_identifier("WING_FLEX_DEV_NEG_STIFF_COEFF".to_owned()),
+            exponent_flex_id: context.get_identifier("WING_FLEX_DEV_STIFF_EXPO_ENA".to_owned()),
+
+            external_accelerations_filtered: LowPassFilter::new(Duration::from_millis(50)),
+        }
+    }
+
+    fn update(
+        &mut self,
+        context: &UpdateContext,
+        lift_forces: &[f64],
+        fuel_masses: [Mass; NODE_NUMBER],
+        external_acceleration_from_plane_body: Acceleration,
+    ) {
+        self.updater_max_step.update(context);
+
+        for cur_time_step in &mut self.updater_max_step {
+            self.external_accelerations_filtered
+                .update(context.delta(), external_acceleration_from_plane_body);
+
+            // Here we artificially move up or down wing root point so that plane movement is communicated to the rest of the wing
+            //      through the flex constraints
+            self.nodes[0].apply_external_offet(
+                Self::PLANE_ACCEL_TO_WING_ROOT_OFFSET_GAIN
+                    * Length::new::<meter>(
+                        self.external_accelerations_filtered
+                            .output()
+                            .get::<meter_per_second_squared>()
+                            .max(-Self::MAX_G_FORCE_IMPACT_APPLIED_ON_WING_ROOT_BY_PLANE_M_S2)
+                            .min(Self::MAX_G_FORCE_IMPACT_APPLIED_ON_WING_ROOT_BY_PLANE_M_S2),
+                    ),
+            );
+
+            // Solving flex physics of n-1 first nodes (N nodes  N-1 Links)
+            for idx in 0..LINK_NUMBER {
+                self.nodes[idx].set_fuel_mass(fuel_masses[idx]);
+
+                self.nodes[idx].apply_force(Force::new::<newton>(lift_forces[idx]));
+
+                self.nodes[idx].update(&context.with_delta(cur_time_step));
+
+                self.flex_constraints[idx].update(
+                    &context.with_delta(cur_time_step),
+                    &mut self.nodes[idx..=idx + 1],
+                );
+            }
+
+            // Don't forget last node to solve as for loop solves only up to n-1 node
+            self.nodes[NODE_NUMBER - 1].set_fuel_mass(fuel_masses[NODE_NUMBER - 1]);
+            self.nodes[NODE_NUMBER - 1]
+                .apply_force(Force::new::<newton>(lift_forces[NODE_NUMBER - 1]));
+            self.nodes[NODE_NUMBER - 1].update(&context.with_delta(cur_time_step));
+        }
+    }
+
+    fn nodes_height_meters(&self) -> [f64; NODE_NUMBER] {
+        let mut all_heights_meters = [0.; NODE_NUMBER];
+
+        for (idx, height) in all_heights_meters
+            .iter_mut()
+            .enumerate()
+            .take(NODE_NUMBER)
+            .skip(1)
+        {
+            *height = self.nodes[idx].position().get::<meter>();
+        }
```

Comment:
This should be the same:
```suggestion
        for (height, node) in all_heights_meters[1..].iter_mut().zip(&self.nodes) {
            *height = node.position().get::<meter>();
        }
```